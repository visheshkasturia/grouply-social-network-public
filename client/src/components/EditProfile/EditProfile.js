/*eslint-disable */
import { useState } from "react";
import { putEditProfile } from "../../apiCalls/putEditProfile";

export default function EditProfile( {user} ) {
  const [inputs, setInputs] = useState({});

  const handleChange = (event) => {
    const name = event.target.name;
    const value = event.target.value;
    setInputs(values => ({...values, [name]: value}))
  }

  const handleSubmit = (event) => {
    event.preventDefault();
    putEditProfile(user._id,inputs);
    setInputs({});
  }

  return (
    <form onSubmit={handleSubmit}>
      <label>Enter Full Name:
      <input 
        type="text" 
        name="fullName" 
        value={inputs.fullName || ""} 
        onChange={handleChange}
      />
      <br />
      </label>
      <label>Enter Username:
      <input 
        type="text" 
        name="userName" 
        value={inputs.userName || ""} 
        onChange={handleChange}
      />
      </label>
      <br />
      <label>Enter Profile Picture URL:
        <input 
          type="text" 
          name="dpURL" 
          value={inputs.dpURL|| ""}
          onChange={handleChange}
        />
        </label>
        <br />
        <button type="submit"> Submit </button> 
    </form>
  )
}
