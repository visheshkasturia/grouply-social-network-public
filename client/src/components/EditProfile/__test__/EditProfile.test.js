import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import EditProfile from '../EditProfile'

it('EditProfile renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<EditProfile />, div);
});

it('EditProfile matches snapshot', () => {
  const tree = renderer.create(<EditProfile />).toJSON();
  expect(tree).toMatchSnapshot();
});