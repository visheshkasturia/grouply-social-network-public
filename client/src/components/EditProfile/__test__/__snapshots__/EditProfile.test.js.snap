// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditProfile matches snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <label>
    Enter Full Name:
    <input
      name="fullName"
      onChange={[Function]}
      type="text"
      value=""
    />
    <br />
  </label>
  <label>
    Enter Username:
    <input
      name="userName"
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
  <br />
  <label>
    Enter Profile Picture URL:
    <input
      name="dpURL"
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
  <br />
  <button
    type="submit"
  >
     Submit 
  </button>
</form>
`;
